src/Core/Primitives.idr

```diff
@@ -147,13 +150,14 @@ strSubstr [NPrimVal fc (I start), NPrimVal _ (I len), NPrimVal _ (Str str)]
     = Just (NPrimVal fc (Str (prim__strSubstr start len str)))
 strSubstr _ = Nothing
 
+
 add : Constant -> Constant -> Maybe Constant
 add (BI x) (BI y) = pure $ BI (x + y)
 add (I x) (I y) = pure $ I (x + y)
 add (B8 x) (B8 y) = pure $ B8 $ (x + y)
 add (B16 x) (B16 y) = pure $ B16 $ (x + y) `mod` (prim__shl_Int 1 16)
 add (B32 x) (B32 y) = pure $ B32 $ (x + y) `mod` (prim__shl_Int 1 32)
```

Comment:
Does it make sense to perform the same refactoring for all the other
`prim__shl__Int 1 N` used in this file?

Reply:
Yes, it would also look nicer to use hexadecimal literals.